Recursive/UnrollMatrix.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @function UnrollMatrix
+ * @description Traverses/Unrolls array of arrays recursively until nothing is left.
+ * @param {Array} matrix - The input array of arrays
+ * @return {Array} matrix - The empty output array => [].
+ * @see https://chortle.ccsu.edu/vectorlessons/vmch13/vmch13_2.html
+ */
+
+const UnrollMatrix = (matrix) => {
+  if (matrix.length === 0) return matrix
+
+  // sweep top to right
+  if (matrix.length > 0) {
+    console.log(...matrix.shift())
```

Comment:
Please pass a visitor function instead of hardcoding `console.log` (or even better, turn the entire function into an iterator).